### PR TITLE
Implement ChatBot and ImageBot for iOS

### PR DESCRIPTION
**Pull request by ewpatton**
Change-Id: I5beea8f2ae60de72afcdf718fdd440195d0d2fe9

### CODE REVIEW

**Comment by ewpatton** [comment]
A few notes:

- The code for computing the edit mask in ImageBot isn't working yet. I'm still investigating.
- ChatBot includes Apikey and maps it to ApiKey for naming consistency. This is already fixed in #2903.
- This should be merged after #2903 but can be tested with ai2-test.

File: appinventor/components-ios/src/ImageBot.swift [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+      let result = UIGraphicsImageRenderer(size: size, format: format).pngData { context in
+        image.draw(in: CGRect(origin: .zero, size: size), blendMode: .copy, alpha: 1.0)
+      }
+      return result
+    }
+    return context.pngRepresentation(of: bitmap, format: .ARGB8, colorSpace: ImageBot.sRGB)
+  }
+
+  func saveImage(_ content: Data) throws -> String {
+    let destination = try FileUtil.getPictureFile("png")
```
[the remark this answers is not in the document]
Comment (daki7711) [reply appinventor/components-ios/src/ImageBot.swift:169]: NVM

[the remark this answers is not in the document]
Comment (ewpatton) [reply appinventor/components-ios/src/ImageBot.swift:169]: OpenAI only returns PNG file content, so this is safe.

**Comment by halatmit** [comment]
This is fine for now when were' using only OpenAI.  But ai expect we'll
also include other options before too long.

== Hal

Hal Abelson
***@***.***
Prof. of Comp. Sci. and Eng.
MIT Dept. of Elec. Eng. and Comp. Sci.

**Comment by mark-friedman** [comment]
> This is fine for now when were' using only OpenAI. But ai expect we'll also include other options before too long.

I think that the input format and the output format can, presumably, be treated separately.  I would argue that simplicity and separation of concerns suggests that the input format be canonicalized and that the proxy make sure that the components receive the canonical format.  Later, we can add the general capability of saving images to other image formats, if we like.

**Comment by halatmit** [comment]
That sounds reasonable. But we should get more experience with those other
platforms before  freezing anything.

== Hal

Hal Abelson
***@***.***
Prof. of Comp. Sci. and Eng.
MIT Dept. of Elec. Eng. and Comp. Sci.

**Comment by mark-friedman** [comment]
> That sounds reasonable. But we should get more experience with those other platforms before freezing anything.

I don't think that the approach implemented here freezes anything, nor do I believe that it makes it difficult to support the sorts of things that I mentioned in my original reply.  I could be wrong, though. Did you have any specific concerns, Hal?

**Comment by halatmit** [comment]
I haven't looked enough at the platforms to know.

== Hal

Hal Abelson
***@***.***
Prof. of Comp. Sci. and Eng.
MIT Dept. of Elec. Eng. and Comp. Sci.

**Review by daki7711 — APPROVED** [review]
LGTM

**Pull request merged**